### PR TITLE
Study creation now uses queue

### PR DESCRIPTION
**Pull request by archendel**

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]
Rapidly tested OK.

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 1]
```diff
@@ -354,27 +345,48 @@ private void deleteFile(@NonNull File file) {
...
             if (caseUuid != null) {
-                String caseFormat = getCaseFormat(caseUuid, basicStudyInfos.getId(), userId);
-                NetworkInfos networkInfos = persistentStore(caseUuid, basicStudyInfos.getId(), userId, importReportUuid);
-                LoadFlowParameters loadFlowParameters = LoadFlowParameters.load();
-                insertStudy(basicStudyInfos.getId(), userId, networkInfos, caseFormat, caseUuid, false, toEntity(loadFlowParameters), importReportUuid);
+                emitStudyCreate(basicStudyInfos.getId(), caseUuid, userId);
             }
         } catch (Exception e) {
             LOGGER.error(e.toString(), e);
+            deleteStudyIfNotCreationInProgress(basicStudyInfos.getId(), userId);
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyService.java:356]: Why delete here but not in other?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 2]
```diff
@@ -354,27 +345,48 @@ private void deleteFile(@NonNull File file) {
...
         } finally {
             deleteFile(caseFile);
-            deleteStudyIfNotCreationInProgress(basicStudyInfos.getId(), userId);
-            LOGGER.trace("Create study '{}' : {} seconds", basicStudyInfos.getId(),
-                    TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime.get()));
         }
     }
 
+    @Bean
+    @Transactional
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyService.java:363]: Doesn't do anything

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 3]
```diff
@@ -303,20 +304,10 @@ public BasicStudyInfos createStudy(UUID caseUuid, String userId, UUID studyUuid)
     }
 
     private void createStudyAsync(UUID caseUuid, String userId, BasicStudyInfos basicStudyInfos) {
-        AtomicReference<Long> startTime = new AtomicReference<>();
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyService.java:306]: Maybe remove the whole method?

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 1 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:356]: it has to be delete in both cases, fixed

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 2 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:363]: If I'm not wrong, it's opening a transaction for the studyEntity creation (line 380)

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 3 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:306]: Should those actions now be made synchronously ?

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 2 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:363]: removed

**Comment by SlimaneAmar** [comment]
Remove `TempFileService`

**Pull request closed without merging**